### ISSUE TITLE
[Bug]: Text should call `initDimensions` when styles change

### ISSUE THREAD

**Issue by ShaMan123** [fabricjs/fabric.js#8975]
I expect the dimensions to update when I set styles because styles affect the size of chars.
I believe a proxy will make this easy to solve.

@asturur @melchiar thoughts?

**Comment by melchiar** [comment]
Does it not already? I had thought styles was included in dimensionsAffectingProps

**Comment by ShaMan123** [comment]
I don't think so and even if they do it is only shallow, meaning that it would recalc if you replace the entire style object, but if you change a prop nothing will trigger

**Comment by asturur** [comment]
yeah because mutation.
You can just invalidate the cache when you set the style, or call `set('style', .... )`
